src/js/angular/core/directives/rdfresourcesearch/rdf-resource-search.directive.js

```diff
@@ -13,13 +13,23 @@ function rdfResourceSearchDirective($rootScope, $timeout,
     templateUrl: 'js/angular/core/directives/rdfresourcesearch/templates/rdfResourceSearchTemplate.html',
     restrict: 'AE',
     link: function ($scope, element) {
+      let autocompleteStatus = false;
 
       function refreshRepositoryInfo() {
         if ($scope.getActiveRepository()) {
           $scope.getNamespacesPromise = RDF4JRepositoriesRestService.getNamespaces(
               $scope.getActiveRepository())
               .success(function () {
-                $scope.getAutocompletePromise = AutocompleteRestService.checkAutocompleteStatus();
+                checkAutocompleteStatus();
+              });
+        }
+      }
+
+      function checkAutocompleteStatus() {
+        if (!autocompleteStatus) {
```

Comment:
If `autocompleteStatus=false` then you are going to check it against the GDB and eventually if it's enabled you rise the flag. So far so good. But what happens if `autocompleteStatus=true` and the user tries to disable the autocomplete? You won't notice it here, will you?

Reply:
Maybe better approach is to find if it's possible to expose the autocomplete status globally from the `AutocompleteCtrl` somehow, e.g. through a statefull service or through the global scope.

Reply:
Refactored code into a service. Added local storage key and watcher. When status is changed it is broadcasted in rootScope.